### DISCUSSION TITLE
SSG path with dynamic path segment?

### DISCUSSION THREAD

**Discussion by aaronadamsCA** [gatsbyjs/gatsby#36407] [Help] [answered]
Can I nest a client route within a collection route using the File System Route API?

I want a route like `/product/{product.id}/variant/[variantId]`. I want to statically generate each product page using the product data, and then get the variant ID at run time.

Is this possible, or do I need to manually implement this using query parameters or a router component?

**Comment by aaronadamsCA** [comment, marked as answer]
Holy crap - this just works!

`src/pages/products/{product.id}/variants/[variantId].tsx`

```tsx
interface VariantPageParams extends PageProps<VariantPageQuery> {
  params: {
    variantId: string;
  };
}

const VariantPage: FC<VariantPageParams> = ({
  data: { product },
  params: { variantId },
}) => {
  if (product == null) {
    throw new Error("Failed to create product page");
  }

  return (
    <div>
      <div>{product.id}</div>
      <div>{variantId}</div>
    </div>
  );
};

export default VariantPage;

export const query = graphql`
  query VariantPage($id: String!) {
    product(id: { eq: $id }) {
      id
    }
  }
`;
```

This should definitely be documented - it's a great way to combine these two features. I haven't yet seen any examples showing the ability to statically generate pages with internal dynamic routing like this, although maybe I was just using the wrong search terms.

**Reply by LekoArts** [reply]
We have it in the example here: https://github.com/gatsbyjs/gatsby/blob/master/examples/route-api/src/pages/parks/%7Bpark.meta__location__type%7D/%5Bname%5D.js

But it's not mentioned in https://www.gatsbyjs.com/docs/reference/routing/file-system-route-api/#example-use-cases. If you want, write something up and help your future self and others :D
